### ISSUE TITLE
CSLAMVC project: CheckObjectRules warning

### ISSUE THREAD

**Issue by grahamblandford** [MarimerLLC/cslaforum#565]
Any idea how I resolve this warning?:

![image](https://user-images.githubusercontent.com/28442774/41049639-62d101de-697f-11e8-80a5-23103bcfdeaa.png)

I don't seem to be able to override it, as it's protected?:

![image](https://user-images.githubusercontent.com/28442774/41049679-7c5a55d8-697f-11e8-99fa-36b15140debc.png)

CSLA version: 4.7.100

**Comment by hurcane** [comment]
There is a good StackOverflow question that addresses your scenario.

[https://stackoverflow.com/questions/6236909/why-cant-we-change-access-modifier-while-overriding-methods-in-c
](url)

You could "hide" the method, but I think that you might want to question why you are trying to do this. If you can explain why you want to make CheckObjectRules public on your business classes, then there might be alternative techniques to achieve your purposes.

**Comment by GillesBer** [comment]
You can't use 'public', keep it 'protected'! The rule subsystem is designed to be managed by the Model. The outside world does not need (it is our understanding of CSLA) to be known by your UI, ...

**Comment by grahamblandford** [comment]
Thanks GillesBer.

I actually pulled the code directly from the CSLAMVC project, so I have no idea why that would have been a public method. Never occurred to me to think "deeper" :/

Thanks again!

**Comment by rockfordlhotka** [comment]
In a server-side scenario it is common to suppress the running of rules while the postback data is being loaded into the object, and then run the rules once after the data's been loaded. There's an interface available for this purpose - don't remember it off the top of my head, but it is used by the CSLA model binder in the Csla.Web.Mvc projects.

**Comment by GillesBer** [comment]
we do use in DataPortal_Fetch {
using (BypassPropertyChecks) {
   //set all properties here, without running each rule 
}
BusinessRules.CheckRules(); //run all rules, taking into account priority to avoid running the ones not relevant at that stage and/or requiring additional I/O

is it what you mean @rockfordlhotka  ?

**Comment by rockfordlhotka** [comment]
No, that is _inside_ the model. I'm referring to the ability to suppress and invoke the rules from _outside_ the model - typically in a controller.

**Comment by jonnybee** [comment]
It is ICheckRules

    public interface ICheckRules
    {
        /// <summary>
        /// Sets value indicating no rule methods will be invoked.
        /// </summary>
        void SuppressRuleChecking();

        /// <summary>
        /// Resets value indicating all rule methods will be invoked.
        /// </summary>
       void ResumeRuleChecking();

        /// <summary>
        /// Invokes all rules for the business type.
        /// </summary>
        void CheckRules();

        /// <summary>
        /// Gets the broken rules collection
        /// </summary>
        /// <returns></returns>
        BrokenRulesCollection GetBrokenRules();
    }
